### ISSUE TITLE
Please upload fix of test_cursor.py to pypi

### ISSUE THREAD

**Issue by cornelinux** [PyMySQL/PyMySQL#466]
I fail to install pymysql in a virtualenv or build an RPM depending on pymysql due to the syntax error in test_cursor.py.

## Current Behavior

<pre>
Compiling /root/rpmbuild/BUILDROOT/privacyidea-2.12-1.el7.centos.x86_64/opt/privacyidea/lib/python2.7/site-packages/pymysql/tests/test_cursor.py ...
  File "/opt/privacyidea/lib/python2.7/site-packages/pymysql/tests/test_cursor.py", line 111
    cursor.executemany(q, [(3, 4), (5, 6)])
         ^
SyntaxError: invalid syntax

error: Bad exit status from /var/tmp/rpm-tmp.usBOF3 (%install)
</pre>

## Possible Solution

Please upload the fix to pypi.

**Comment by methane** [comment]
I'm sorry for inconvinience.
But your report doesn't include how to reproduce.

I **can** install in venv.
```
$ bin/pip -V
pip 8.1.2 from /Users/inada-n/tmp/v/lib/python3.5/site-packages (python 3.5)
[inada-n@PC2197:~/tmp/v]
$ bin/pip install pymysql
Collecting pymysql
  Downloading PyMySQL-0.7.3-py2.py3-none-any.whl (77kB)
    100% |████████████████████████████████| 81kB 4.5MB/s
Installing collected packages: pymysql
Successfully installed pymysql-0.7.3
```

**Comment by methane** [comment]
I can install from tar.gz also.
Syntax error shouldn't prevent install.
```
$ bin/pip install pymysql --no-binary=:all:
Collecting pymysql
  Downloading PyMySQL-0.7.3.tar.gz (67kB)
    100% |████████████████████████████████| 71kB 5.1MB/s
Installing collected packages: pymysql
  Running setup.py install for pymysql ... done
Successfully installed pymysql-0.7.3
```

**Comment by cornelinux** [comment]
Thanks a lot for the reply.
Sorry for the missing information!

I am building an RPM on CentOS7. So I am using python 2.7.5

On a CentOS7 system you can clone this repo
https://github.com/NetKnights-GmbH/centos7
and run "make buildrpm".

Roughly it will issue these commands:

    virtualenv /opt/privacyidea
    source /opt/privacyidea/bin/activate
    pip install privacyidea==2.12
    pip install pymysql

When installing pymysql the pip install exists with an error, which breaks the complete RPM building process.

<pre>
(privacyidea)[root@localhost rpmbuild]# pip install pymysql
Downloading/unpacking pymysql
  Downloading PyMySQL-0.7.3.tar.gz (67kB): 67kB downloaded
  Running setup.py egg_info for package pymysql
    
Installing collected packages: pymysql
  Running setup.py install for pymysql
      File "/opt/privacyidea/lib/python2.7/site-packages/pymysql/tests/test_cursor.py", line 111
        cursor.executemany(q, [(3, 4), (5, 6)])
             ^
    SyntaxError: invalid syntax
    
    
  Could not find .egg-info directory in install record for pymysql
Successfully installed pymysql
Cleaning up...
</pre>

**Comment by methane** [comment]
> Successfully installed pymysql

You succeeded to install PyMySQL.  Don't you??

**Comment by cornelinux** [comment]
If you install it manually it is installed.
But the %build section of the RPM spec file nevertheless breaks when the install process throws this systax error.
...maybe due to the byte-compile of the RPM package.

# Reproduce

1. install CentOS 7
2. run
    git clone https://github.com/NetKnights-GmbH/centos7
3. cd centos7
4. make buildrpm

<pre>
+ pip install pymysql
Requirement already satisfied (use --upgrade to upgrade): pymysql in /opt/privacyidea/lib/python2.7/site-packages
Cleaning up...
+ pip install pymysql_sa
Downloading/unpacking pymysql_sa
  Downloading pymysql_sa-1.0.tar.gz
  Running setup.py egg_info for package pymysql-sa
    
Requirement already satisfied (use --upgrade to upgrade): pymysql>=0.3 in /opt/privacyidea/lib/python2.7/site-packages (from pymysql_sa)
Requirement already satisfied (use --upgrade to upgrade): SQLAlchemy>=0.6.4 in /opt/privacyidea/lib/python2.7/site-packages (from pymysql_sa)
Installing collected packages: pymysql-sa
  Running setup.py install for pymysql-sa
    
  Could not find .egg-info directory in install record for pymysql_sa
Successfully installed pymysql-sa
Cleaning up...
+ rm -fr /opt/privacyidea/lib/python2.7/site-packages/authmodules
+ rm -fr /opt/privacyidea/lib/privacyidea/authmodules
+ exit 0
Executing(%install): /bin/sh -e /var/tmp/rpm-tmp.3FBTmt
+ umask 022
+ cd /root/rpmbuild/BUILD
+ '[' /root/rpmbuild/BUILDROOT/privacyidea-2.12-1.el7.centos.x86_64 '!=' / ']'
+ rm -rf /root/rpmbuild/BUILDROOT/privacyidea-2.12-1.el7.centos.x86_64
++ dirname /root/rpmbuild/BUILDROOT/privacyidea-2.12-1.el7.centos.x86_64
+ mkdir -p /root/rpmbuild/BUILDROOT
+ mkdir /root/rpmbuild/BUILDROOT/privacyidea-2.12-1.el7.centos.x86_64
+ mkdir -p /root/rpmbuild/BUILDROOT/privacyidea-2.12-1.el7.centos.x86_64/opt/
+ cp -r /opt/privacyidea /root/rpmbuild/BUILDROOT/privacyidea-2.12-1.el7.centos.x86_64/opt/
+ '[' x86_64 = noarch ']'
+ QA_CHECK_RPATHS=1
+ case "${QA_CHECK_RPATHS:-}" in
+ /usr/lib/rpm/check-rpaths

+ /usr/lib/rpm/check-buildroot
+ /usr/lib/rpm/redhat/brp-compress
+ /usr/lib/rpm/redhat/brp-strip /usr/bin/strip
+ /usr/lib/rpm/redhat/brp-strip-comment-note /usr/bin/strip /usr/bin/objdump
+ /usr/lib/rpm/redhat/brp-strip-static-archive /usr/bin/strip
+ /usr/lib/rpm/brp-python-bytecompile /usr/bin/python 1
Compiling /root/rpmbuild/BUILDROOT/privacyidea-2.12-1.el7.centos.x86_64/opt/privacyidea/lib/python2.7/site-packages/pymysql/tests/test_cursor.py ...
  File "/opt/privacyidea/lib/python2.7/site-packages/pymysql/tests/test_cursor.py", line 111
    cursor.executemany(q, [(3, 4), (5, 6)])
         ^
SyntaxError: invalid syntax

error: Bad exit status from /var/tmp/rpm-tmp.3FBTmt (%install)


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.3FBTmt (%install)
make: *** [buildrpm] Error 1
</pre>

**Comment by methane** [comment]
Wow, RPM is crazy.
Python package may contain harmless syntax error.  For example, `_xxx_py3.py` module which imported only on Python 3 cause syntax error while bytecompiling on Python 2.
Additionally, pip bytecompiles python modules as necessary. Why rpm tries bytecompile again?

While I'll release next version in this week, I recommend you to use pip simply (without rpm) for managing Python modules.

**Comment by cornelinux** [comment]
Thanks. It is building a binary package, not a noarch. Maybe because of this.
Thanks a o lot for the upcoming release!
I realized version 0.6.6 would also still work for me...

**Comment by methane** [comment]
I released 0.7.4 now.

**Comment by cornelinux** [comment]
thx a lot.
And thx again for this package. Great job!
